swap_meet/vendor.py

```diff
@@ -1,2 +1,115 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        '''
+        Removes the matching item from the `inventory`
+        Returns the item that was removed
+        If there is no matching item in the `inventory`, return `False`
+        '''
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+        
+    def get_by_id(self, id):
+        '''
+        This method takes one argument: an integer, representing an `Item`'s `id`
+        Returns the item with a matching `id` from the inventory
+        If there is no matching item in the `inventory`, return`None`
+        '''
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+
+        self.inventory.remove(my_item)
+        self.inventory.append(their_item)
+        other_vendor.inventory.remove(their_item)
+        other_vendor.inventory.append(my_item)
+
+        return True
+    
+    def swap_first_item(self, other_vendor):
+        '''
+        Removes the first item from instance's `inventory`, and adds the friend's first item
+        Removes the first item from the friend's `inventory`, and adds the instances first item
+        It returns `True`
+        If either itself or the friend have an empty `inventory`, return `False`
+
+        '''
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        else:
+            item_to_swap = self.inventory.pop(0)
+            other_item_to_swap = other_vendor.inventory.pop(0)
+            self.inventory.append(other_item_to_swap)
+            other_vendor.inventory.append(item_to_swap)
+        return True
+    
+    def get_by_category(self, category):
+        '''
+        Returns a list of objects in the inventory with a category
+        If there are no items in the `inventory` that match the category argument, 
+        Return an empty list
+        '''
+        items = []
+        for item in self.inventory:
+            if item.get_category() == category:
+                items.append(item)
+        return items
```

Comment:
Notice this pattern of: 
```python
result_list = []
for element in source_list:
    if some_condition(element):
        result_list.append(element)
```

can be rewritten using a list comprehension as:
```python
result_list = [element for element in source_list if some_condition(element)]
```

Which here would look like:
```python
items = [item for item in self.inventory if item.get_category() == category]
return items
```

At first, this may seem more difficult to read, but comprehensions are a very common python style, so I encourage y’all to try working with them!